### DISCUSSION TITLE
Package build is failing with --extra-index-url in the requirements.txt

### DISCUSSION THREAD

**Discussion by pranayvooredi-tec** [pypa/setuptools#3688] [Q&A] [answered]
Hi there, 

I am trying to build a private pypi package using Gitlab with setuptools for one of our internal projects.
**This package depends on another gitlab private package. So, I have included that package name with extra index url in the requirements.txt

My requirement.txt is as follows:
```
requests==2.28.1
--extra-index-url https://gitlab-ci-token:<personal_access_token>@git.examples.com/api/v4/projects/987/packages/pypi/simple redishandler
```

in setup.py, I have added install_requires as below:
```
with open('../requirements.txt') as f:
    required = f.read().splitli**nes()

setuptools.setup(
    name="tokenutility",
    version="0.0.15",
    author="Pranay vooredi",
    author_email="abc@xyz.com",
    description="Token Utility package",
    packages=["tokenutility"],
    classifiers=[
        "Programming Language :: Python :: 3",
        "License :: OSI Approved :: MIT License",
        "Operating System :: OS Independent",
    ],
    python_requires='>=3.6',
    install_requires=required
)
```

But package build is failing with below error:
```
error in tokenutility setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers; Parse error at "'--extra-'": Expected W:(abcd...)
```

Can anyone help me how to achieve this package build.

Thanks in Advance

**Comment by abravalheri** [comment, marked as answer]
Hi @pranayvooredi-tec, please note that this is [intentionally not supported by `setuptools`](https://setuptools.pypa.io/en/latest/userguide/dependency_management.html). All dependency strings must follow the syntax specified in [PEP 508](https://peps.python.org/pep-0508/), which implies no `pip`-specific command line options.


I also believe that PyPI will not let you submit a package that uses dependencies in private indexes or 3rd party websites
